src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelStateDictionary.cs

```diff
@@ -184,14 +178,13 @@ public ModelValidationState ValidationState
                 {
                     throw new ArgumentNullException(nameof(value));
                 }
-                _innerDictionary[key] = value;
-            }
-        }
 
-        // For unit testing
-        internal IDictionary<string, ModelStateEntry> InnerDictionary
-        {
-            get { return _innerDictionary; }
+                var entry = GetOrAddNode(key);
+                Count += entry.Visible ? 0 : 1;
+                ErrorCount += value.Errors.Count - entry.Errors.Count;
```

Comment:
This might need to be changed. I assumed `ErrorCount` kept track of all errors, but it's specifically errors added via `AddModelError`. So this change is incorrect.